woocommerce/woocommerce-ios
# Add-Ons: Update ProductAddOn decoding to support heading type

# Why 

I noticed that a while ago my test site started to fail when decoding global add-ons (add-on groups). I couldn't find any significant change in the add-ons extension(Also I don't understand PHP well) but the issue seems to be that the **Heading** add-on type is not returning all the properties that it should return, hence making the decoding to fail.

# How

To fix that I'm adding a custom decoding, where if a property is not present or fails to decode(`Ints`are sent as empty `strings`) I'm falling back to a default `zero` value.

# Testing Steps

- Install the add-ons extension, set up an add-ons group and a heading add-on (or login to my test store)
 <img width="2004" alt="Screen Shot 2021-11-03 at 3 04 59 PM" src="https://user-images.githubusercontent.com/562080/140190536-828559e5-3089-491c-b592-dd056327083c.png">
 
 - Launch the app and make sure add-on groups decode doesn't fail (no error printed in the console)

 
Update release notes:
